keepalived/check/ipvswrapper.c

```diff
@@ -682,8 +688,14 @@ ipvs_group_remove_entry(virtual_server_t *vs, virtual_server_group_entry_t *vsge
 	if (!is_vsge_alive(vsge, vs)) {
 		if (vsge->range)
 			ipvs_group_range_cmd(IP_VS_SO_SET_DEL, &srule, NULL, vsge);
-		else
+		else {
+			if (vsge->addr.ss_family == AF_INET6)
+				inet_sockaddrip6(&vsge->addr, &srule.nf_addr.in6);
+			else 
+				srule.nf_addr.ip = inet_sockaddrip4(&vsge->addr);
+
```

Comment:
The srule will have been set by the call(s) to ipvs_talk()/ipvs_group_range_cmd() with IP_VS_SO_SET_DELDEST in the LIST_FOREACH block above. The only way this would not happen is if there were no real servers, but that would be a configuration error.

Reply:
Indeed if the real servers are all died,  this would not happen.

Reply:
But in fact, I think maybe we just delete the judgement of  vsge->range, whether it is zero or not , it will work well : )